packages/block-library/src/template-part/edit/index.js

```diff
@@ -24,6 +24,9 @@ export default function TemplatePartEdit( {
 
 	// Resolve the post ID if not set, and load its post.
 	const postId = useTemplatePartPost( _postId, slug, theme );
+	useEffect( () => {
+		setAttributes( { postId } );
+	}, [ postId ] );
```

Comment:
This seems to fix it.  However, it means that when we first load a template supplied by a theme, the template and all of its template parts will be marked as 'dirty' immediately since they have all been given explicit `postId`s.  I dont think this is necessarily bad behavior, just different than before. 🤔

Reply:
interesting. I think this behavior also existed at some point in the past (I think @epiqueras changed that somehow)

Reply:
I don't think this behavior is acceptable. We made an effort to remove it because it's confusing.

Reply:
> We made an effort to remove it because it's confusing.

True, definitely confusing if we are prompting the user to save changes to something they haven't made changes to.  🤔

Reply:
Alternative idea - 61fa1bf - also set the `postId` attribute in the name-panel when the slug is changed.  This will prevent it from disappearing when renamed, allow the template to dirty to save the proper id for its child, and not give us any issues with dirty loading. 🤔

Reply:
I think there needs to be a clear distinction between Template Parts that are being referenced from the theme, and those that are customized by the user. Because once you start editing a Template Part, you're actually editing a newly-created duplicate, not the one from the theme. If your theme updates, the Template Part won't change with it, because you're no longer using the theme's version of the Template Part. But this is never conveyed clearly to the user.

I think theme-provided Template Parts should be in a sort of "read-only" mode by default, with an explicit "Customize/Edit" button that, when pressed, will copy it and create the editable version. Right now it's not clear at all whether a Template Part is still being referenced from a theme or not, and I think this is the source of a lot of the confusing behavior/logic.

Reply:
That makes sense to me for sure. Imo, we can move forward with the fixes here, and then come up with a design for that interaction as well.

Reply:
> If your theme updates, the Template Part won't change with it

I could be wrong, but now that you mention this I think this may still be the case even if the user has not edited the template part?  On first use an auto-draft is created from the theme's .html files with the slug name, that is what is being used by the editor.  If the auto-draft with that slug/theme already exists, I don't think we create a new one and overwrite it.  So if the themes template part .html is updated, the editor would still be looking for the auto-draft by slug which was created with the outdated version. 🤔

Reply:
> I think if we handle it this way for template parts we should probably do the same for templates as well?

Yes, that sounds like the right behavior to me.